spec/unit/provider/package/dnfmodule_spec.rb

```diff
@@ -0,0 +1,175 @@
+      end
+
+      it "should install the default stream and flavor" do
+        resource[:ensure] = :present
+        expect(provider).to receive(:execute).with(array_including('baz'))
+        provider.install
+      end
+
+      it "should install a specific stream" do
+        resource[:ensure] = '9.6'
+        expect(provider).to receive(:execute).with(array_including('baz:9.6'))
+        provider.install
+      end
+
+      it "should install a specific flavor" do
+        resource[:ensure] = :present
+        resource[:flavor] = 'minimal'
+        expect(provider).to receive(:execute).with(array_including('baz/minimal'))
+        provider.install
+      end
+
+      it "should install a specific flavor and stream" do
+        resource[:ensure] = '9.6'
+        resource[:flavor] = 'minimal'
+        expect(provider).to receive(:execute).with(array_including('baz:9.6/minimal'))
+        provider.install
+      end
+    end
+
+    context "when ensuring a specific version on top of another stream" do
+      before do
+        provider.instance_variable_get('@property_hash')[:ensure] = '9.6'
+      end
+
+      it "should remove existing packages and reset the module stream before installing" do
+        resource[:ensure] = '10'
+        expect(provider).to receive(:execute).thrice.with(array_including(/remove|reset|install/))
+        provider.install
+      end
+    end
+  end
+
+  context "parsing the output of module list --installed" do
+    before { allow(described_class).to receive(:command).with(:dnf).and_return(dnf_path) }
+
+    it "returns an array of installed modules" do
+      expect(Puppet::Util::Execution).to receive(:execpipe)
+        .with("/usr/bin/dnf module list --installed -d 0 -e 1")
+        .and_yield(packages)
+
+      installed_packages = described_class.instances
+      expect(installed_packages[0].properties).to eq( { :ensure => "2.8", :name => "gimp", :flavor => "devel", :provider => :dnfmodule })
+      expect(installed_packages[1].properties).to eq( { :ensure => "10.3", :name => "mariadb", :flavor => "client", :provider => :dnfmodule })
+      expect(installed_packages[2].properties).to eq( { :ensure => "10", :name => "nodejs", :flavor => "minimal", :provider => :dnfmodule })
+      expect(installed_packages[3].properties).to eq( { :ensure => "5.26", :name => "perl", :flavor => "minimal", :provider => :dnfmodule })
+      expect(installed_packages[4].properties).to eq( { :ensure => "10", :name => "postgresql", :flavor => "server", :provider => :dnfmodule })
+      expect(installed_packages[5].properties).to eq( { :ensure => "3.6", :name => "python36", :flavor => "common", :provider => :dnfmodule })
+      expect(installed_packages[6].properties).to eq( { :ensure => "rhel8", :name => "rust-toolset", :flavor => "common", :provider => :dnfmodule })
+      expect(installed_packages[7].properties).to eq( { :ensure => "1.10", :name => "subversion", :flavor => "server", :provider => :dnfmodule })
```

Comment:
Generally is better to have a single expect per test case(easier to track failures, this test has like 8 reasons to fail), and you can also use `key: value` hash syntax here.

Reply:
do you know how I can compact all the cases in a single expect call? this is also how it's done in rpm_spec.rb